### PR TITLE
Added Azure.Defender.Storage.SensitiveData

### PR DESCRIPTION
**Pull request by BenjaminEngeset**
## PR Summary

Part of #2207 

Added Azure.Defender.Storage.SensitiveData rule.

## PR Checklist

- [x] PR has a meaningful title
- [x] Summarized changes
- [x] Change is not breaking
- [x] This PR is ready to merge and is not **Work in Progress**
- **Rule changes**
  - [x] Unit tests created/ updated
  - [x] Rule documentation created/ updated
  - [x] Link to a filed issue
  - [x] [Change log](https://github.com/Azure/PSRule.Rules.Azure/blob/main/docs/CHANGELOG-v1.md) has been updated with change under unreleased section
- **Other code changes**
  - [ ] Unit tests created/ updated
  - [ ] Link to a filed issue
  - [ ] [Change log](https://github.com/Azure/PSRule.Rules.Azure/blob/main/docs/CHANGELOG-v1.md) has been updated with change under unreleased section

### CODE REVIEW

**Comment by BenjaminEngeset** [comment]
Hi @BernieWhite! What do you think?

File: docs/en/rules/Azure.Defender.Storage.SensitiveData.md [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+
+- Reduce the likelihood of data breaches
+- Detect exposure events
+- Detect suspicious activities on resources containing sensitive data 
+
+When enabling sensitive data threat detection, the sensitive data categories include built-in sensitive information types (SITs) in the default list of Microsoft Purview. This will affect the alerts received from Defender for Storage, storage or containers that are found with these SITs are marked as containing sensitive data.
+
+It is also possible to customize the Data Sensitivity Discovery for a organization, you can create custom sensitive information types (SITs) and connect to your organizational settings.
+
+Sensitive data threat detection in Microsoft Defender for Storage can be enabled at the subscription level and by doing so ensures all storage accounts in the subscription will be protected, including future ones.
```
Comment (BernieWhite) [inline docs/en/rules/Azure.Defender.Storage.SensitiveData.md:35]: When don't need to include everything from the docs, just enough to understand what the feature is and why they should enable it.

```suggestion
Sensitive data threat detection is an additional security feature for Microsoft Defender for Storage.
When enabled Defender for Storage provides alerts when sensitive data is discovered.

The sensitive data threat detection capability helps teams:

- Identity where sensitive data is stored.
- Detect possible security incidents resulting is data exposure.

When enabling sensitive data threat detection, the sensitive data categories include built-in sensitive information types (SITs) in the default list of Microsoft Purview. 
It is possible to customize the Data Sensitivity Discovery for a organization, by creating custom sensitive information types (SITs).

Sensitive data threat detection in Microsoft Defender for Storage can be enabled at the subscription level and by doing so ensures all storage accounts in the subscription will be protected, including future ones.
```

**Review by BernieWhite — CHANGES_REQUESTED** [review]
Thanks @BenjaminEngeset. Minor suggestion on the docs, otherwise should be ok.

**Comment by BenjaminEngeset** [comment]
@BernieWhite Agreed, thanks for identifying and correcting. I have updated it now.

**Review by BernieWhite — APPROVED** [review]
Thanks @BenjaminEngeset. All good to merge.

**Pull request merged**